src/ng/rootScope.js

```diff
@@ -1180,7 +1180,10 @@ function $RootScopeProvider() {
         return function() {
           var indexOfListener = namedListeners.indexOf(listener);
           if (indexOfListener !== -1) {
-            namedListeners[indexOfListener] = null;
+            // Use delete in the hope of the browser deallocating the memory for the array entry,
```

Comment:
> ...in the hope of the browser deallocating...

Why wouldn't it? :grin:

Reply:
I think there should be a more descriptive comment here, e.g. "depending on browser implementations, delete should deallocate the memory"

Reply:
I'm not sure if it "should" though. Is there anything that actually says it should? I think browsers (some? all?) do this just to be more efficient. I kind of like the "in the hope of" so I'm having trouble finding a better description...

Reply:
Did we ever find a conclusive answer to the actual implementation in specific browsers? Or is this based on benchmarks? Either info could be listed here and then maybe a note that we assume other browsers do it as well.

Reply:
Based on benchmarks Chrome and FF seem to both use a more memory efficient array implementation for sparse arrays (which I [noted here](https://github.com/angular/angular.js/issues/16135#issuecomment-334986679)). But I'm not sure about all browsers, or the details of when Chrome and FF do it, and I'm assuming there is no actual standard for this.